src/shared/modules/loader/services/loader.service.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { LoaderState } from '../loader.model';
+@Injectable({
+  providedIn: 'root'
+})
+export class LoaderService {
+  private loaderSubject = new Subject<LoaderState>();
```

Comment:
replace by state plz

Reply:
Loader service is made as an independent module to copy it across applications. Hence it should not be tied to application specific states